### ISSUE TITLE
Deactivating "Check Spelling" does not remove visual indicators

### ISSUE THREAD

**Issue by HeyMyian** [vkbo/novelWriter#1414]
Not sure if bug, but this seems like unusual behavior. If I want to completely deactivate the visual indicators of the spell checking, I have to also close the document and reopen it.

**Comment by vkbo** [comment]
The visual result of the spell checking is generated by the syntax highlighter. Toggling it should trigger a new refresh of the highlighting for the whole document, at least it used to, but I can look into whether there is a bug somewhere in the pipeline.

The decision to highlight is made by the editor, and the user interaction bounces around a bit in the code, so it may fail somewhere in some cases. A few issues are known, for instance with a misconfigured dictionary setting.

**Comment by vkbo** [comment]
Tested and confirmed. The code to refresh spellchecking is called, but it actually only runs if spell checking is enabled, which it isn't if you turn *off* spell checking. Basically, it will refresh when enabling spell checking, but not the reverse.

**Comment by HeyMyian** [comment]
> Edit: You don't have to close/reopen actually. You can just rerun the spell checker with the toggle off.

That does not work for me. Situation is: Spell check is on. I disable spell check. I rerun spell check. --> Nothing changes, the squiggly lines remain.

**Comment by vkbo** [comment]
> That does not work for me. Situation is: Spell check is on. I disable spell check. I rerun spell check. --> Nothing changes, the squiggly lines remain.

Yeah, I just realised that the same if-condition I removed also blocks that. It will work after the fix.

The fix *only* fixes it for small documents though, but that as is expected. For big documents, all running of the spell checker is done manually due to performance reasons. The definition of a big document is set in Preferences, but defaults to 800 kB. You'll only be able to reach that limit if you put a full novel in a single document, so it's a non-issue in most cases. The main exception is if you import a single document novel and want to split it up.
